pkg/avancement/service_manager.go

```diff
@@ -52,19 +52,26 @@ const fromBranch = "master"
 // It uses a Git cache to checkout the code to, and will copy the environment
 // configuration for the `fromURL` to the `toURL` in a named branch.
 func (s *ServiceManager) Promote(serviceName, fromURL, toURL, newBranchName string) error {
-	source, err := s.checkoutSourceRepo(fromURL, fromBranch)
-	if err != nil {
-		return err
-	}
 	destination, err := s.checkoutDestinationRepo(toURL, newBranchName)
 	if err != nil {
 		return fmt.Errorf("failed to checkout repo: %w", err)
 	}
-
-	copied, err := git.CopyService(serviceName, source, destination)
-
-	if err != nil {
-		return fmt.Errorf("failed to copy service: %w", err)
+	copied := []string{}
+	if fromSoruceRepo(fromURL) {
+		source, err := s.repoFactory("https://source/sourceorg/sourcerepo", fromURL) 
```

Comment:
Maybe this should be `""` rather than something that looks like a repo?

Reply:
Agreed, this line is confusing. What kinds of values does `fromURL` have in these cases? We expect the source to be a clone of a functional git repository in our primary use case but do we have to know the url it was cloned from? Is that what fromURL is?

Reply:
The fromURL is a directory path from the command line.  The `repoFactory` needs a n url format string.  I'll change the `repoFactory` to take "".